java/ql/lib/semmle/code/java/JDK.qll

```diff
@@ -197,6 +203,48 @@ class TypeFile extends Class {
   TypeFile() { this.hasQualifiedName("java.io", "File") }
 }
 
+/**
+ * A taint step from an update of the `bytes[]` parameter in an override of the `InputStream.read` method
+ * to a class instance expression of the type extending `InputStream`.
+ *
+ * This models how a subtype of `InputStream` could be tainted by the definition of its methods, which will
+ * normally only happen in anonymous classes.
+ */
+private class InputStreamWrapperAnonymousStep extends AdditionalTaintStep {
```

Comment:
Should we move these additional taint step definitions somewhere into the frameworks subfolder instead of having them here in `JDK.qll`? We already have various flow models for jdk in there in a couple of different files (e.g. `Stream.qll`).

Reply:
Good point, done in e752861.